### ISSUE TITLE
Minecart Upgrade Kit change to make it more compatible with modded carts

### ISSUE THREAD

**Issue by direwolf420** [bluemagic123/tModLoader#3597]
### Do you intend to personally contribute/program this feature?

No

### I would like to see this change made to improve my experience with

Mod capability as a Modder

### Description

The [Minecart Upgrade Kit](https://terraria.wiki.gg/wiki/Minecart_Upgrade_Kit) sets fixed values for movement stats on carts. This is fine in vanilla because every cart has the exact same movement stats. In modded it proves problematic because changes to those default stats will not reflect with the Kit.

I propose a solution that will instead scale the default values via a factor, or specify the override values directly (both 
customizable as fields in `MountData`). This also allows for balancing.

### What does this proposal attempt to solve or improve?
Currently, if you want to have your changes scale properly, you need to use `ModPlayer.PostUpdateRunSpeeds`, check for the mount, check for the kit being active, and then adjust the relevant movement fields on the player.

### Which (other) solutions should be considered?

See Description

**Comment by TechnicalitiesDevelopment** [comment]
I’ll do it

**Comment by TechnicalityCreations** [comment]
I wouldn’t mind implementing it. I would suggest that we take the latter option and just add a parameter to the `Mount.MountData` type, as different base speeds exist for different types of vanilla minecarts, making it hard to create a factor.

**Comment by JavidPack** [comment]
Good work, there are some things that will have to be changed to prevent breaking existing mods, but we'll fix those when we finalize the design.

I'm not sure if the approach presented here is flexible enough for what modders want, so I've made a discussion thread on our Discord in the development forum (https://discord.com/channels/103110554649894912/1294026615910699160). If you want to participate in that discussion as it happens, ping me on Discord so I can add you to the thread, I didn't see a user with your name when I checked.
